pkg/kubelet/lifecycle/predicate.go

```diff
@@ -230,6 +230,17 @@ func removeMissingExtendedResources(pod *v1.Pod, nodeInfo *schedulerframework.No
 			podCopy.Spec.Containers[i].Resources.Requests[rName] = rQuant
 		}
 	}
+	for i, c := range pod.Spec.InitContainers {
+		podCopy.Spec.InitContainers[i].Resources.Requests = make(v1.ResourceList)
```

Comment:
should this be done for EphemeralContainers too?

Reply:
I think that seems unnecessary because EphemeralContainers  are not allowed to set resources.
[what-is-an-ephemeral-container](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/#what-is-an-ephemeral-container)

Reply:
great point!